### PR TITLE
Issue #331 Determining directory names within archive dynamically

### PR DESCRIPTION
**Pull request by hferentschik**
Addresses issue #331

### CODE REVIEW

File: pkg/util/github/github.go [hunk 1]
```diff
@@ -235,6 +248,28 @@ func DownloadOpenShiftReleaseBinary(binaryType OpenShiftBinaryType, osType minis
 	return nil
 }
 
+func listDirExcluding(dir string, excludeRegexp string) ([]string, error) {
```
Comment (budhrg) [inline pkg/util/github/github.go]: `func listDirExcluding(dir, excludeRegexp string)` format here?

File: pkg/util/archive/archive.go [hunk 2]
```diff
@@ -70,15 +65,13 @@ func Untar(tarball, targetDir string) error {
...
 		path := filepath.Join(targetDir, header.Name)
-		info := header.FileInfo()
-		if info.IsDir() {
-			if err = os.MkdirAll(path, info.Mode()); err != nil {
-				return err
-			}
-			continue
+
+		// tar.Next() will externally only iterate files, so we might have to create intermediate directories here
+		if err = os.MkdirAll(filepath.Dir(path), 0770); err != nil {
```
Comment (budhrg) [inline pkg/util/archive/archive.go]: :+1:

**Comment by budhrg** [comment]
LGTM :+1: if you take care one best practice related comment :wink:

File: pkg/util/github/github.go (hunk 1 above)
Comment (hferentschik) [inline pkg/util/github/github.go]: format? what do you mean?

Comment (praveenkumar) [inline pkg/util/github/github.go]: @budhrg I also don't understand what do you mean format here. function definition seems good to me.

Comment (budhrg) [inline pkg/util/github/github.go]: @hferentschik @praveenkumar , I mean as per Go, if you have same datatype of parameters in the function(like for `dir` and `excludeRegexp` have `string` as datatype) then you keep the datatype of leftmost parameter and omit rest as : 
```
func listdirExcluding(dir, excludeRegexp string) ([]string, error) {
....
}
```

Its a common idiom in Go.
Sorry I didn't elaborate my comment previously.
```

Comment (hferentschik) [inline pkg/util/github/github.go]: hmm, one can for sure do so. I am not sure whether this is part of best practices or idiomatic in Go. Personally I am not a big fan.

**Pull request closed without merging**